> **提示**：欢迎参加“调查数据集”项目！引用段会添加这种提示，帮助你制定调查方法。提交项目之前，最后浏览一下报告，将这一段删除，以保持报告简洁。首先，需要双击这个 Markdown 框，将标题更改为与数据集和调查相关的标题。

# 项目：研究票房高的电影具有的特点

## 目录
<ul>
<li><a href="#intro">简介</a></li>
<li><a href="#wrangling">数据整理</a></li>
<li><a href="#eda">探索性数据分析</a></li>
<li><a href="#conclusions">结论</a></li>
</ul>

<a id='intro'></a>
## 简介

我们从Kaggle获取了从1960年到2015年之间的1万多部电影的TMBd数据， 这包括电影的受欢迎度（popularity），预算（budget），票房（revenue），导演，演职人员（cast），电影类型（genres）等。在这片报告中，我们想要从3个方面来探索票房（revenue）高的电影所具有的特点。第一，票房与预算之间的关系，提高预算能否保证电影的高票房。第二，TMDb的评分与票房之间的关系，是不是高票房的电影就会有更好的评分。第三，电影类型与票房之间的关系，特定类型的电影是否更容易获得高票房。


In [52]:
# 用这个框对你计划使用的所有数据包设置
#   导入语句。
import pandas as pd
import numpy as np

# 务必包含一个‘咒语’，以便将你的视图
#   与 notebook 保持一致。关于更多信息，请访问这个网页：
#   http://ipython.readthedocs.io/en/stable/interactive/magics.html
%matplotlib inline

<a id='wrangling'></a>
## 数据整理

> **提示**：在这一段报告中载入数据，检查简洁度，然后整理和清理数据集，以进行分析。务必将步骤仔细归档，并确定清理决策是否正确。

### 常规属性

In [49]:
# 加载数据并打印几行。进行几项操作，检查数据
#   类型，以及是否有缺失数据或错误数据。
movies= pd.read_csv('data/tmdb-movies.csv')
movies.head()

id    imdb_id  popularity     budget     revenue  \
0  135397  tt0369610   32.985763  150000000  1513528810   
1   76341  tt1392190   28.419936  150000000   378436354   
2  262500  tt2908446   13.112507  110000000   295238201   
3  140607  tt2488496   11.173104  200000000  2068178225   
4  168259  tt2820852    9.335014  190000000  1506249360   

                 original_title  \
0                Jurassic World   
1            Mad Max: Fury Road   
2                     Insurgent   
3  Star Wars: The Force Awakens   
4                     Furious 7   

                                                cast  \
0  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
1  Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
2  Shailene Woodley|Theo James|Kate Winslet|Ansel...   
3  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
4  Vin Diesel|Paul Walker|Jason Statham|Michelle ...   

                                            homepage          director  \
0                      http://www.jurassicworld.com/   Colin Trevorrow   
1                        http://www.madmaxmovie.com/     George Miller   
2     http://www.thedivergentseries.movie/#insurgent  Robert Schwentke   
3  http://www.starwars.com/films/star-wars-episod...       J.J. Abrams   
4                           http://www.furious7.com/         James Wan   

                         tagline      ...       \
0              The park is open.      ...        
1             What a Lovely Day.      ...        
2     One Choice Can Destroy You      ...        
3  Every generation has a story.      ...        
4            Vengeance Hits Home      ...        

                                            overview runtime  \
0  Twenty-two years after the events of Jurassic ...     124   
1  An apocalyptic story set in the furthest reach...     120   
2  Beatrice Prior must confront her inner demons ...     119   
3  Thirty years after defeating the Galactic Empi...     136   
4  Deckard Shaw seeks revenge against Dominic Tor...     137   

                                      genres  \
0  Action|Adventure|Science Fiction|Thriller   
1  Action|Adventure|Science Fiction|Thriller   
2         Adventure|Science Fiction|Thriller   
3   Action|Adventure|Science Fiction|Fantasy   
4                      Action|Crime|Thriller   

                                production_companies release_date vote_count  \
0  Universal Studios|Amblin Entertainment|Legenda...       6/9/15       5562   
1  Village Roadshow Pictures|Kennedy Miller Produ...      5/13/15       6185   
2  Summit Entertainment|Mandeville Films|Red Wago...      3/18/15       2480   
3          Lucasfilm|Truenorth Productions|Bad Robot     12/15/15       5292   
4  Universal Pictures|Original Film|Media Rights ...       4/1/15       2947   

   vote_average  release_year    budget_adj   revenue_adj  
0           6.5          2015  1.379999e+08  1.392446e+09  
1           7.1          2015  1.379999e+08  3.481613e+08  
2           6.3          2015  1.012000e+08  2.716190e+08  
3           7.5          2015  1.839999e+08  1.902723e+09  
4           7.3          2015  1.747999e+08  1.385749e+09  

[5 rows x 21 columns]

In [76]:
movies[movies['budget']== 0].head()

id    imdb_id  popularity  budget   revenue  \
30  280996  tt3168230    3.927333       0  29355203   
36  339527  tt1291570    3.358321       0  22354572   
72  284289  tt2911668    2.272044       0     45895   
74  347096  tt3478232    2.165433       0         0   
75  308369  tt2582496    2.141506       0         0   

                    original_title  \
30                      Mr. Holmes   
36                          Solace   
72                Beyond the Reach   
74          Mythica: The Darkspore   
75  Me and Earl and the Dying Girl   

                                                 cast  \
30  Ian McKellen|Milo Parker|Laura Linney|Hattie M...   
36  Abbie Cornish|Jeffrey Dean Morgan|Colin Farrel...   
72  Michael Douglas|Jeremy Irvine|Hanna Mangan Law...   
74  Melanie Stone|Kevin Sorbo|Adam Johnson|Jake St...   
75  Thomas Mann|RJ Cyler|Olivia Cooke|Connie Britt...   

                                             homepage  \
30                       http://www.mrholmesfilm.com/   
36                                                NaN   
72                                                NaN   
74          http://www.mythicamovie.com/#!blank/wufvh   
75  http://www.foxsearchlight.com/meandearlandthed...   

                   director  \
30              Bill Condon   
36            Afonso Poyart   
72  Jean-Baptiste Léonetti   
74            Anne K. Black   
75      Alfonso Gomez-Rejon   

                                              tagline      ...       \
30                            The man behind the myth      ...        
36  A serial killer who can see your future, a psy...      ...        
72                                                NaN      ...        
74                                                NaN      ...        
75           A Little Friendship Never Killed Anyone.      ...        

                                             overview runtime  \
30  The story is set in 1947, following a long-ret...     103   
36  A psychic doctor, John Clancy, works with an F...     101   
72  A high-rolling corporate shark and his impover...      95   
74  When Teela’s sister is murdered and a powerf...     108   
75  Greg is coasting through senior year of high s...     105   

                      genres  \
30             Mystery|Drama   
36       Crime|Drama|Mystery   
72                  Thriller   
74  Action|Adventure|Fantasy   
75              Comedy|Drama   

                                 production_companies release_date vote_count  \
30  BBC Films|See-Saw Films|FilmNation Entertainme...      6/19/15        425   
36  Eden Rock Media|FilmNation Entertainment|Flynn...       9/3/15        474   
72                                      Furthur Films      4/17/15         81   
74                           Arrowstorm Entertainment      6/24/15         27   
75                                  Indian Paintbrush      6/12/15        569   

    vote_average  release_year  budget_adj   revenue_adj  
30           6.4          2015         0.0  2.700677e+07  
36           6.2          2015         0.0  2.056620e+07  
72           5.5          2015         0.0  4.222338e+04  
74           5.1          2015         0.0  0.000000e+00  
75           7.7          2015         0.0  0.000000e+00  

[5 rows x 21 columns]

In [77]:
movies[movies['revenue']== 0].head()

id    imdb_id  popularity    budget  revenue  \
48  265208  tt2231253    2.932340  30000000        0   
67  334074  tt3247714    2.331636  20000000        0   
74  347096  tt3478232    2.165433         0        0   
75  308369  tt2582496    2.141506         0        0   
92  370687  tt3608646    1.876037         0        0   

                    original_title  \
48                       Wild Card   
67                        Survivor   
74          Mythica: The Darkspore   
75  Me and Earl and the Dying Girl   
92        Mythica: The Necromancer   

                                                 cast  \
48  Jason Statham|Michael Angarano|Milo Ventimigli...   
67  Pierce Brosnan|Milla Jovovich|Dylan McDermott|...   
74  Melanie Stone|Kevin Sorbo|Adam Johnson|Jake St...   
75  Thomas Mann|RJ Cyler|Olivia Cooke|Connie Britt...   
92  Melanie Stone|Adam Johnson|Kevin Sorbo|Nicola ...   

                                             homepage             director  \
48                                                NaN           Simon West   
67                          http://survivormovie.com/       James McTeigue   
74          http://www.mythicamovie.com/#!blank/wufvh        Anne K. Black   
75  http://www.foxsearchlight.com/meandearlandthed...  Alfonso Gomez-Rejon   
92          http://www.mythicamovie.com/#!blank/y9ake        A. Todd Smith   

                                        tagline     ...       \
48  Never bet against a man with a killer hand.     ...        
67           His Next Target is Now Hunting Him     ...        
74                                          NaN     ...        
75     A Little Friendship Never Killed Anyone.     ...        
92                                          NaN     ...        

                                             overview runtime  \
48  When a Las Vegas bodyguard with lethal skills ...      92   
67  A Foreign Service Officer in London tries to p...      96   
74  When Teela’s sister is murdered and a powerf...     108   
75  Greg is coasting through senior year of high s...     105   
92  Mallister takes Thane prisoner and forces Mare...       0   

                      genres  \
48      Thriller|Crime|Drama   
67     Crime|Thriller|Action   
74  Action|Adventure|Fantasy   
75              Comedy|Drama   
92  Fantasy|Action|Adventure   

                                 production_companies release_date vote_count  \
48  Current Entertainment|Lionsgate|Sierra / Affin...      1/14/15        481   
67  Nu Image Films|Winkler Films|Millennium Films|...      5/21/15        280   
74                           Arrowstorm Entertainment      6/24/15         27   
75                                  Indian Paintbrush      6/12/15        569   
92  Arrowstorm Entertainment|Camera 40 Productions...     12/19/15         11   

    vote_average  release_year    budget_adj  revenue_adj  
48           5.3          2015  2.759999e+07          0.0  
67           5.4          2015  1.839999e+07          0.0  
74           5.1          2015  0.000000e+00          0.0  
75           7.7          2015  0.000000e+00          0.0  
92           5.4          2015  0.000000e+00          0.0  

[5 rows x 21 columns]

In [55]:
#check if there is duplicated data
sum(movies.duplicated())

1

In [50]:
# check data integrity 
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10866 entries, 0 to 10865
Data columns (total 21 columns):
id                      10866 non-null int64
imdb_id                 10856 non-null object
popularity              10866 non-null float64
budget                  10866 non-null int64
revenue                 10866 non-null int64
original_title          10866 non-null object
cast                    10790 non-null object
homepage                2936 non-null object
director                10822 non-null object
tagline                 8042 non-null object
keywords                9373 non-null object
overview                10862 non-null object
runtime                 10866 non-null int64
genres                  10843 non-null object
production_companies    9836 non-null object
release_date            10866 non-null object
vote_count              10866 non-null int64
vote_average            10866 non-null float64
release_year            10866 non-null int64
budget_adj              1

从上面可以看到，我们的数据集中一共有10866部电影，每部电影共有21个属性，在这里有一些数据我们是不关心的，这包括：id，imdb_id，homepage，tagline，keywords，runtime，production_companies，release_date, release_year， 这些列将会在我们的数据清理阶段被剔除出去。同时我们可以发现有23部电影没有被定义他们的类型（genres），这些数据需要我们经过探索来决定是否需要留下他们，以及如何处理他们。

In [47]:
movies[movies["genres"].isnull()]

id    imdb_id  popularity  budget  revenue  \
424    363869  tt4835298    0.244648       0        0   
620    361043  tt5022680    0.129696       0        0   
997    287663        NaN    0.330431       0        0   
1712    21634  tt1073510    0.302095       0        0   
1897    40534  tt1229827    0.020701       0        0   
2370   127717  tt1525359    0.081892       0        0   
2376   315620  tt1672218    0.068411       0        0   
2853    57892  tt0270053    0.130018       0        0   
3279    54330  tt1720044    0.145331       0        0   
4547   123024  tt2305700    0.520520       0        0   
4732   139463  tt2084977    0.235911       0        0   
4797   369145        NaN    0.167501       0        0   
4890   126909  tt2219564    0.083202       0        0   
5830   282848  tt2986512    0.248944       0        0   
5934   200204  tt2808968    0.067433       0        0   
6043   190940  tt2797242    0.039080       0        0   
6530   168891  tt0818519    0.092724       0        0   
8234    56804  tt0114844    0.028874       0        0   
8614    65595  tt0117880    0.273934       0        0   
8878    92208  tt0250593    0.038045       0        0   
9307   141859  tt0097446    0.094652       0        0   
9799    48847  tt0193716    0.175008       0        0   
10659    4255  tt0065904    0.344172    5000        0   

                                          original_title  \
424                                       Belli di papà   
620                                   All Hallows' Eve 2   
997                 Star Wars Rebels: Spark of Rebellion   
1712                                   Prayers for Bobby   
1897              Jonas Brothers: The Concert Experience   
2370                                     Freshman Father   
2376                       Doctor Who: A Christmas Carol   
2853                                           Vizontele   
3279                                       아기와 나   
4547   London 2012 Olympic Opening Ceremony: Isles of...   
4732                                       The Scapegoat   
4797                             Doctor Who: The Snowmen   
4890                          Cousin Ben Troop Screening   
5830                  Doctor Who: The Time of the Doctor   
5934                                        Prada: Candy   
6043                                      Bombay Talkies   
6530                                         Saw Rebirth   
8234                                     Viaggi di nozze   
8614                          T2 3-D: Battle Across Time   
8878                     Mom's Got a Date With a Vampire   
9307                                           Goldeneye   
9799                                         The Amputee   
10659                      The Party at Kitty and Stud's   

                                                    cast  \
424    Diego Abatantuono|Matilde Gioli|Andrea Pisani|...   
620                                                  NaN   
997    Freddie Prinze Jr.|Vanessa Marshall|Steve Blum...   
1712   Ryan Kelley|Sigourney Weaver|Henry Czerny|Dan ...   
1897   Nick Jonas|Joe Jonas|Kevin Jonas|John Lloyd Ta...   
2370   Britt Irvin|Merrilyn Gann|Barbara Tyson|Anthon...   
2376   Matt Smith|Karen Gillan|Arthur Darvill|Michael...   
2853   Yılmaz Erdoğan|Demet Akbag|Altan Erkekli|Cem...   
3279            Jang Keun-suk|Song Ha-yoon|Kim Jeong-Nan   
4547   Queen Elizabeth II|Mike Oldfield|Kenneth Brana...   
4732   Andrew Scott|Jodhi May|Eileen Atkins|Matthew R...   
4797   Matt Smith|Jenna Coleman|Richard E. Grant|Ian ...   
4890                                   Jason Schwartzman   
5830                            Matt Smith|Jenna Coleman   
5934            Peter Gadiot|Rodolphe Pauly|Léa Seydoux   
6043   Aamir Khan|Rani Mukerji|Randeep Hooda|Saqib Sa...   
6530   Whit Anderson|Stan Kirsch|Jeff Shuter|George W...   
8234   Carlo Verdone|Claudia Gerini|Veronica Pivetti|...   
8614   Arnold Schwarzenegger|Linda Hamilton|Edward Fu...   
8

In [79]:
movies[movies["vote_average"].isnull()]

Empty DataFrame
Columns: [id, imdb_id, popularity, budget, revenue, original_title, cast, homepage, director, tagline, keywords, overview, runtime, genres, production_companies, release_date, vote_count, vote_average, release_year, budget_adj, revenue_adj]
Index: []

[0 rows x 21 columns]

> **提示**：_不应_在每个框中进行太多操作。可以自由创建框，进行数据探索。在这个项目中，可以在初始 notebook 中进行大量探索操作。不要求对其进行组织，但务必仔细阅读备注，理解每个代码框的用途。完成分析之后，可以创建 notebook 副本，在其中去除多余数据，组织步骤，从而形成结构连贯、紧密的报告。

> **提示**：务必向你的读者告知你在调查中采取的步骤。在每个代码框或每组相关代码框后面，用 markdown 框对前面的框中的调查结果向读者进行说明。尽量做到这一点，以便读者理解后续框中的内容。

### 从数据中去除掉无用的列

In [ ]:
# 讨论数据结构和需要解决的任何问题之后，
#   在本段的第二部分进行这些清理步骤。


<a id='eda'></a>
## 探索性数据分析

> **提示**：整理和清理数据之后，现在可以进行探索。计算统计值，创建视图，解决你在简介段提出的研究问题。建议采用系统化方法。一次探索一个变量，然后探索变量之间的关系。

### 研究问题 1（替换这个标题名称！）

In [ ]:
# 用这个代码框和其它代码框探索数据。务必添加
#   Markdown 框，以便归档你的观察和调查结果。


### 研究问题 2（替换这个标题名称！）

In [ ]:
# 继续探索数据，解决你的附加研究问题。
#   如果有其它问题要调查，
#   请根据需要添加更多标题。


<a id='conclusions'></a>
## 结论

> **提示**：最后，总结你的调查结果。确保了解探索结果的限制。如果尚未进行任何统计检验，不要做出任何统计结论。切记不要根据相互关系推导出因果关系！

> **提示**：如果对报告满意，应将其副本保存为 HTML 或 PDF 形式。导出报告之前请检查一遍，确保报告流程完整。应删除所有类似的“提示”引用段，以保持报告简洁。还需要查看课程结尾的项目提交页的项目审阅规范。

> 如需将报告导出到工作区，应运行下面的代码框。如果正确，会返回代码 0，工作区目录下会生成 .html 文件（单击左上角的 jupyter 图标）。也可以通过 **文件** > **下载为** 子菜单下载 html 报告，然后手动上传到工作区目录。完成之后，可以单击右下角的“提交项目”，提交你的项目。恭喜！

In [ ]:
from subprocess import call
call(['python', '-m', 'nbconvert', 'Investigate_a_Dataset.ipynb'])